## Importing packages -

In [27]:
import numpy as np
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

## Data -

In [4]:
x = np.array([5, 15, 25, 35, 45, 55]).reshape((-1, 1))  #.reshape() on x because this array is required to be two-dimensional, or to be more precise, to have one column and as many rows as necessary
y = np.array([5, 20, 14, 32, 22, 38])

In [5]:
print(x)

[[ 5]
 [15]
 [25]
 [35]
 [45]
 [55]]


In [6]:
print(y)

[ 5 20 14 32 22 38]


## Fitting a model-

In [7]:
model = LinearRegression()

In [8]:
model.fit(x,y)

LinearRegression()

In [9]:
model = LinearRegression().fit(x,y)

## Getting the result -

In [11]:
r_sq = model.score(x,y)

In [13]:
print('coeeficient of determinant', r_sq)

coeeficient of determinant 0.715875613747954


In [14]:
# attributes of model are .intercept_, which represents the coefficient, 𝑏₀ and .coef_, which represents 𝑏₁:
print('intercept:', model.intercept_)

intercept: 5.633333333333329


In [17]:
print('slope:',model.coef_)

slope: [0.54]


The value 𝑏₀ = 5.63 (approximately) illustrates that your model predicts the response 5.63 when 𝑥 is zero. The value 𝑏₁ = 0.54 means that the predicted response rises by 0.54 when 𝑥 is increased by one.



## Predicting - 

In [19]:
y_pred = model.predict(x)
print('Predicted Response:', y_pred, sep='\n')

Predicted Response:
[ 8.33333333 13.73333333 19.13333333 24.53333333 29.93333333 35.33333333]
